Makefile

```
@@ -0,0 +1,14 @@
+build:
+	cargo build
+
+test:
+	cargo test -- --test-threads=1
+
+run-local:
+	cargo run
+
+lint:
+	cargo clippy
```

I feel that Makefiles, particularly with such a clean structure as this one, are great. I've seen several other task runners, but I'm not sure there is another standard one. I've seen `cargo make` (https://github.com/sagiegurari/cargo-make) but I'm not sure it is worth to explore at this time. Maybe if this Makefile starts getting too big?

The only thing I'd suggest here is:
```suggestion
	cargo clippy  --all-features --tests
```
so it also checks the tests.